src/live_restore/live_restore_fs.c

```diff
@@ -729,35 +868,17 @@ __live_restore_fh_read(
 int
 __wti_live_restore_fs_fill_holes(WT_FILE_HANDLE *fh, WT_SESSION *wt_session)
 {
-/*
- * Holes can be large, potentially the size of an entire file. When we find a large hole we'll read
- * it in 4KB chunks.
- */
-#define WT_LIVE_RESTORE_READ_SIZE ((size_t)(4 * WT_KILOBYTE))
-    WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
-    WT_LIVE_RESTORE_HOLE_NODE *hole;
-
-    char buf[WT_LIVE_RESTORE_READ_SIZE];
-    lr_fh = (WT_LIVE_RESTORE_FILE_HANDLE *)fh;
-
-    while ((hole = lr_fh->destination.hole_list_head) != NULL) {
-        __wt_verbose_debug3((WT_SESSION_IMPL *)wt_session, WT_VERB_FILEOPS,
-          "Found hole in %s at %" PRId64 "-%" PRId64 " during background migration. ", fh->name,
-          hole->off, WT_EXTENT_END(hole));
+    WT_DECL_RET;
+    bool finished = false;
 
-        /* If panic is set on the connection stop doing work. */
+    while (!finished) {
+        WT_WITH_LIVE_RESTORE_EXTENT_LIST_WRITE_LOCK((WT_SESSION_IMPL *)wt_session,
```

Comment:
I am not a fan of this casting but with the scope of this PR being huge already I'd prefer to fix it in another ticket.